packages/core/upload/server/src/services/upload.ts

```diff
@@ -553,9 +553,48 @@ export default ({ strapi }: { strapi: Core.Strapi }) => {
     });
   }
 
+  /**
+   * Uploads a file by path, buffer or stream programmaticaly
+   * @public
+   */
+  async function uploadFile(file: string | Buffer | fs.ReadStream, options: FileInfo) {
```

Comment:
Do we specifically want to only handle fs streams? Or should we allow any kind of readable stream? (`NodeJS.ReadableStream`)